src/librustc_codegen_llvm/builder.rs

```diff
@@ -518,19 +622,37 @@ impl Builder<'a, 'll, 'tcx> {
         }
     }
 
-    pub fn nonnull_metadata(&self, load: &'ll Value) {
+    fn nonnull_metadata(&mut self, load: &'ll Value) {
```

Comment:
Similar here - in fact, we can do this in one place - the LLVM `!nonnull` is just `!range` but for pointers, and only when the range excludes `0` (which is what the caller checks).